public/components/common/custom-search-bar/custom-search-bar.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState, useEffect, useLayoutEffect } from 'react';
+
+import { getIndexPattern } from '../../overview/mitre/lib'
+import { Filter } from '../../../../../../src/plugins/data/public/';
+import { FilterMeta, FilterState, FilterStateStore } from '../../../../../../src/plugins/data/common';
+
+import {
+    EuiFlexGroup,
+    EuiFlexItem,
+    EuiSwitch,
+} from '@elastic/eui';
+
+//@ts-ignore
+import { getDataPlugin } from '../../../kibana-services';
+import { KbnSearchBar } from '../../kbn-search-bar';
+import { TimeRange, Query } from '../../../../../../src/plugins/data/common';
+import { Combobox } from './components'
+
+export const CustomSearchBar = ({ filtersValues, ...props }) => {
+
+    const KibanaServices = getDataPlugin().query;
+    const filterManager = KibanaServices.filterManager;
+    const timefilter = KibanaServices.timefilter.timefilter;
+    const indexPattern = getIndexPattern();
+    const [filterParams, setFilterParams] = useState({
+        filters: filterManager.getFilters().map(({ meta: { removable, ...restMeta }, ...rest }) => ({ ...rest, meta: restMeta })) || [],
+        query: { language: 'kuery', query: '' },
+        time: timefilter.getTime(),
+    });
+    const defaultSelectedOptions = () => {
+        const array = []
+        filtersValues.forEach(item => {
+            array[item.key] = []
+        })
+
+        return array
+    }
+    const [isLoading, setLoading] = useState(false);
+    const [avancedFiltersState, setAvancedFiltersState] = useState(false);
+    const [selectedOptions, setSelectedOptions] = useState(defaultSelectedOptions);
+
+    useEffect(() => {
+        let filterSubscriber = filterManager.getUpdates$().subscribe(() => {
+            const newFilters = filterManager.getFilters();
+            onFiltersUpdated(newFilters)
+            return () => {
+                filterSubscriber.unsubscribe();
+            };
+        });
+    }, []);
+
+    const onQuerySubmit = (payload: { dateRange: TimeRange, query: Query }) => {
+        const { query, dateRange } = payload;
+        const filters = { query, time: dateRange, filters: filterParams.filters };
+        setFilterParams(filters);
+
+    }
+
+    const onFiltersUpdated = (filters: Filter[]) => {
+        const { query, time } = filterParams;
+        const updatedFilterParams = { query, time, filters };
+        setFilterParams(updatedFilterParams);
```

Comment:
This could be done by having `filters` use it's own `useState` or with
`setFilterParams({...filterParams, filters})`